File: 0201_0300/0277/0277.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    a, b, c = (int(x) for x in f.readline().split())
    s = ""
    if (a, b, c) == (0, 0, 0):
        s = "0"
    else:
        if a:
            s = str(a) + "+" if b > 0 else str(a)
        if b:
            s += str(b) + "x" if abs(b) > 1 else "-x" if b < 0 else "x"
        if c > 0 and (a or b):
            s += "+"
        if c:
            s += str(c) + "y" if abs(c) > 1 else "-y" if c < 0 else "y"
    q.write(s)

File: 0201_0300/0277/0277.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int a, b, c;

	fscanf(f, "%d%d%d", &a, &b, &c);
	if (!a && !b && !c)
		putc('0', q);
	else
	{
		if (a)
			fprintf(q, b > 0 ? "%d+" : "%d", a);
		if (b)
			fprintf(q, b > 1 || b < -1 ? "%dx" : b < 0 ? "-x" : "x", b);
		if (c > 0 && (a != 0 || b != 0))
			putc('+', q);
		if (c)
			fprintf(q, c > 1 || c < -1 ? "%dy" : c < 0 ? "-y" : "y", c);
	}
	return 0;
}
